File: dotnet/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example2/File.cs
namespace CompositeDesignPattern.Example2;

public class File : IFileSystemItem
{
	public double Size { get; set; }
	public string Extension { get; set; }

	public File(string name, string path, string extension, double size) : base(name, path)
	{
		Extension = extension;
		Size = size;
	}

	public override double GetSize() => Size;
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example2/src/File.ts
import { FileSystemItem } from './FileSystemItem';

export class File extends FileSystemItem {
  constructor(
    name: string,
    path: string,
    public readonly extension: string,
    public readonly size: number,
  ) { super(name, path); }

  getSize(): number { return this.size; }
}
